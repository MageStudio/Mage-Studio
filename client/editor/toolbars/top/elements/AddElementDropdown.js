import React from 'react';
import { Dropdown, Button, Menu } from 'antd';
import { PlusOutlined } from '@ant-design/icons';

import style from '../toolbar.module.scss';
const { Item, SubMenu } = Menu;

const getAddMenu = (onItemSelect) => (
    <Menu onClick={onItemSelect}>
        <Item title='model'>
            model
        </Item>
        <SubMenu title="mesh">
            <Item key='cube' title='cube'>cube</Item>
            <Item key='sphere' title='sphere'>sphere</Item>
            <Item key='cylinder' title='cylinder'>cylinder</Item>
        </SubMenu>
        <SubMenu title="sound">
            <Item>sorry</Item>
        </SubMenu>
        <SubMenu title="light">
            <Item key='ambient' title='ambient'>ambient light</Item>
            <Item key='sun' title='sun'>sun light</Item>
        </SubMenu>
    </Menu>
);

const AddElementDropdown = ({ onElementSelection = f => f }) => {
    const onItemSelect = ({ key }) => {
        onElementSelection(key);
    };

    return (
        <div className={style['controls-settings-group']}>
            <Dropdown
                overlay={getAddMenu(onItemSelect)}
                trigger={['click']}
                placement={'topLeft'}>
                <Button className={style['space-settings-button']}>
                    <PlusOutlined/>Add
                </Button>
            </Dropdown>
        </div>
    )
};

export default AddElementDropdown;