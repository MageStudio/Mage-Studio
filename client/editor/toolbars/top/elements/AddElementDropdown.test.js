import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';

import AddElementDropdown from './AddElementDropdown';
import sinon from 'sinon';

describe('AddElementDropdown', () => {

    describe('render', () => {

        it('should render properly', () => {
            const component = shallow(<AddElementDropdown />);
            expect(toJSON(component)).toMatchSnapshot();
        });
    });
});