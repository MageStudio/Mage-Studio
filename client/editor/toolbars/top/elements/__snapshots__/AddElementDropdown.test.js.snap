// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddElementDropdown render should render properly 1`] = `
<div
  className="controls-settings-group"
>
  <Dropdown
    mouseEnterDelay={0.15}
    mouseLeaveDelay={0.1}
    overlay={
      <Menu
        onClick={[Function]}
      >
        <MenuItem
          title="model"
        >
          model
        </MenuItem>
        <SubMenu
          title="mesh"
        >
          <MenuItem
            title="cube"
          >
            cube
          </MenuItem>
          <MenuItem
            title="sphere"
          >
            sphere
          </MenuItem>
          <MenuItem
            title="cylinder"
          >
            cylinder
          </MenuItem>
        </SubMenu>
        <SubMenu
          title="sound"
        >
          <MenuItem>
            sorry
          </MenuItem>
        </SubMenu>
        <SubMenu
          title="light"
        >
          <MenuItem
            title="ambient"
          >
            ambient light
          </MenuItem>
          <MenuItem
            title="sun"
          >
            sun light
          </MenuItem>
        </SubMenu>
      </Menu>
    }
    placement="topLeft"
    trigger={
      Array [
        "click",
      ]
    }
  >
    <Button
      className="space-settings-button"
    >
      <ForwardRef(PlusOutlined) />
      Add
    </Button>
  </Dropdown>
</div>
`;
